src/org/ods/services/OpenShiftService.groovy

```diff
@@ -193,8 +194,23 @@ class OpenShiftService {
             // a rollout just before we wanted to start it. In that case, we
             // do not need to fail.
             def newVersion = getLatestVersion(name)
+            logger.warn("Rollout - deployment-last #${version} / latest #${newVersion} has errored out: ERR ${ex}")
             if (newVersion > version) {
                 logger.debug("Deployment #${newVersion} has been started by another process")
+            } else if (newVersion == version) {
+                // there is a corner case - where multiple triggers fired, e.g mono-repo and caused
+                // n deployments (in various states, e.g. one cancelled, one running)
+                // we "assume" the running on to be the right one - and bubble
+                // that one up - as surrounding code will check anyway for matching of
+                // image sha's etc...
+                def replicationController = "${name}-${newVersion}"
+                def rolloutStatus = getRolloutStatus(replicationController)
+                if (rolloutStatus != 'running') {
+                    logger.debug("Deployment #${version} / latest #${newVersion} has errored out: ERR ${ex}, status: ${rolloutStatus}")
+                    throw ex
+                } else {
+                    logger.debug("New Deployment #${newVersion} already running")
```

Comment:
were you able to test this against the issue? I also think that if the rollout is running, it is OK to continue. However, I have two open questions in my head:

1. could there be a case where a rollout is running for a previous version, and we are not picking it up? that would suggest we should cancel, and then restart?
2. I don't really understand how the error happens yet so not sure if this actually solves the issue ...

Reply:
> 1. could there be a case where a rollout is running for a previous version, and we are not picking it up? that would suggest we should cancel, and then restart?

as we verify anyway the image sha's later - I think that is fine - but indeed - that would be option two (yet complex) because another trigger could fire in between, and we have the same issue... :;D